core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -1322,11 +1411,20 @@ void skip0() {
     public static class Remote extends ProcessTree implements Serializable {
         private final IProcessTree proxy;
 
+        @Deprecated
         public Remote(ProcessTree proxy, Channel ch) {
             this.proxy = ch.export(IProcessTree.class,proxy);
```

Comment:
Seems weird to me to not just delegate to the new constructor with new argument, instead code duplication. Or am I missing something?

Reply:
It was mostly about the semantics of the arguments describing the behavior, the original constructor doesn't actually know, so I was attempting to limit the assumption of behavior to the base class only.  Maybe wasn't the best plan, ideally it'd be since to remove the previous constructor but I wasn't sure if that'd be an API issue or not.